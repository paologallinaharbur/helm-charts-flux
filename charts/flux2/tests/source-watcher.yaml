suite: test source-watcher deployment
templates:
  - source-watcher.yaml
tests:
  - it: should be empty if source-watcher is not enabled
    asserts:
      - hasDocuments:
          count: 0
    set:
      sourceWatcher.create: false
  - it: should be empty by default
    asserts:
      - hasDocuments:
          count: 0
    set: {}
  - it: should have kind Deployment for k8s >= 1.19
    capabilities:
      majorVersion: 1
      minorVersion: 19
    set:
      sourceWatcher.create: true
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Deployment
      - isAPIVersion:
          of: apps/v1
  - it: should set additional container args
    set:
      sourceWatcher.create: true
      sourceWatcher.container.additionalArgs:
        - --testlabel1=testvalue1
    asserts:
      - contains:
          path: spec.template.spec.containers[0].args
          content: --testlabel1=testvalue1
  - it: should match snapshot with default values when enabled
    set:
      sourceWatcher.create: true
    asserts:
      - matchSnapshot: {}
  - it: should set imagePullPolicy to Always
    set:
      sourceWatcher.create: true
      sourceWatcher.imagePullPolicy: Always
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: Always
  - it: should imagePullPolicy to be default value IfNotPresent
    set:
      sourceWatcher.create: true
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
  - it: should use default cluster domain when null
    set:
      clusterDomain: null
      sourceWatcher.create: true
    asserts:
      - contains:
          path: spec.template.spec.containers[0].args
          content: --storage-adv-addr=source-watcher.$(RUNTIME_NAMESPACE).svc.cluster.local.
  - it: should use custom cluster domain
    set:
      sourceWatcher.create: true
      clusterDomain: custom.domain
    asserts:
      - contains:
          path: spec.template.spec.containers[0].args
          content: --storage-adv-addr=source-watcher.$(RUNTIME_NAMESPACE).svc.custom.domain.
  - it: should override securityContext
    set:
      sourceWatcher.create: true
      sourceWatcher.podSecurityContext:
        runAsUser: 2000
      sourceWatcher.securityContext:
        runAsUser: 3000
    asserts:
      - equal:
          path: spec.template.spec.securityContext
          value:
            runAsUser: 2000
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            runAsUser: 3000
  - it: should not add runAsUser if distro.openshift=true
    set:
      distro:
        openshift: true
    asserts:
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - ALL
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
