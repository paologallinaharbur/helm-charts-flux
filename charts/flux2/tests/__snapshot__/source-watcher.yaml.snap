should match snapshot with default values when enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: source-watcher
        app.kubernetes.io/instance: NAMESPACE
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: flux
        app.kubernetes.io/version: 2.7.2
        control-plane: controller
        helm.sh/chart: flux2-2.17.1
      name: source-watcher
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: source-watcher
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            prometheus.io/port: "8080"
            prometheus.io/scrape: "true"
          labels:
            app: source-watcher
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - --events-addr=http://notification-controller.$(RUNTIME_NAMESPACE).svc.cluster.local.
                - --watch-all-namespaces=true
                - --log-level=info
                - --log-encoding=json
                - --enable-leader-election
                - --storage-path=/data
                - --storage-adv-addr=source-watcher.$(RUNTIME_NAMESPACE).svc.cluster.local.
              env:
                - name: RUNTIME_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
              image: ghcr.io/fluxcd/source-watcher:v2.0.2
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: healthz
              name: manager
              ports:
                - containerPort: 9090
                  name: http
                  protocol: TCP
                - containerPort: 8080
                  name: http-prom
                  protocol: TCP
                - containerPort: 9440
                  name: healthz
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources:
                limits: {}
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 65534
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /data
                  name: data
                - mountPath: /tmp
                  name: tmp
          securityContext:
            fsGroup: 1337
          serviceAccountName: source-watcher-controller
          terminationGracePeriodSeconds: 10
          volumes:
            - emptyDir: {}
              name: data
            - emptyDir: {}
              name: tmp
